pkg/trie/cache/trie_cache.go

```diff
@@ -0,0 +1,11 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package cache
+
+type TrieCache interface {
+	GetValue(key []byte) []byte
+	SetValue(key []byte, value []byte)
+	GetNode(key []byte) []byte
+	SetNode(key []byte, value []byte)
```

Comment:
Do we need to mirror the same interface as the parity trie-db `TrieCache` trait? I see that they don't have an analagous `SetNode` method, but rather have a  `get_or_insert_node` method.

Reply:
We can do that, but I want to keep it simple. The `get_or_insert_node` function requires a callback function if the value isn't present. I prefer the Go way, which is more verbose, and explicitly call the `SetNode` method if we need to add that node.